activerecord/test/cases/view_test.rb

```diff
@@ -47,6 +47,11 @@ def test_table_exists
     assert @connection.table_exists?(view_name), "'#{view_name}' table should exist"
   end
 
+  def test_views_ara_valid_data_sources
+    view_name = Ebook.table_name
+    assert @connection.view_exists?(view_name), "'#{view_name}' should be a data source"
```

Comment:
`assert @connection.data_source_exists?(view_name), "'#{view_name}' should be a data source"`  ??